data/sea/54-zebra.h

```
@@ -155,73 +142,186 @@ static ierror_msg_t zebra_translate
-                    int64_t start = data.a.n[index]   - data.a.n[0];
-                    int64_t len   = data.a.n[index+1] - data.a.n[index];
-                    offset = zebra_translate_table (mempool, start, len, (void**)(strings + i), &data.a.table);
-                }
-                *dst = strings;
-                return offset;
+        case ZEBRA_COLUMN_ENUM: {
+            int64_t column_count = data._enum.columns.count;
+            int64_t offset = zebra_translate_column_tags
+                ( column_count
+                , dst
+                , data._enum.tags );
+            for (int64_t i = 0; i != column_count; ++i) {
+                offset += zebra_translate_column
+                    ( mempool
+                    , elem_start
+                    , elem_count
+                    , dst + offset
+                    , data._enum.columns.columns + i );
             }
+            return offset;
+        }
+
+        case ZEBRA_COLUMN_STRUCT: {
+            int64_t offset = 0;
+            int64_t column_count = data._struct.columns.count;
+            for (int64_t i = 0; i != column_count; ++i) {
+                offset += zebra_translate_column
+                    ( mempool
+                    , elem_start
+                    , elem_count
+                    , dst + offset
+                    , data._struct.columns.columns + i );
+            }
+            return offset;
+        }
+
+        case ZEBRA_COLUMN_NESTED: {
+            return zebra_translate_column_nested
+                ( mempool
+                , elem_start
+                , elem_count
+                , dst
+                , data._nested.indices
+                , &data._nested.table );
+        }
+
+        case ZEBRA_COLUMN_REVERSED: {
+            return zebra_translate_column
+                ( mempool
+                , elem_start
+                , elem_count
+                , dst
+                , &data._reversed );
+        }
+
     }
```

There's a few places where we are switching on a `tag` and not handling the case where we don't know what the tag is. We need to return an error code in that case, or at the very least print an error and call `exit(1)`.